ecopod/sendRawTransactionConditional/proposal.md

```diff
@@ -52,22 +56,42 @@ Verticalization is possible in the proposed solution by configuring the allowlis
 
     * **Runtime Shutoff**
 
-        Through a new admin endpoint exposed through op-node, the sequencer can reject all calls to this endpoint, similar to the `admin_(start|stop)Sequencer` endpoints. If a proxy implementing these validation rules are setup to run on seperate hosts from the sequencer, this runtime switch must be readable such that these requests can be terminated without reaching the sequencer.
+        Through an admin endpoint or environment variable in the proxy, all calls can be rejected without reaching the sequencer. When rejecting, the json-rpc request will return an error with the code -32000 and "unavailable" message. Bundlers should be setup to observe this error state and fallback to `sendRawTransaction` until further communication.
 
-        As long as a single bundler supports a fallback to `eth_sendRawTransaction`, 4337 liveness should remain OK
+        As long as a single bundler supports a fallback to `eth_sendRawTransaction`, shared-mempool 4337 liveness should remain OK
 
     * **Global Rate Limit**
 
-        Rate limit how many conditional txs can be sent over a configured time period. Like with the runtime shutoff, if these validation rules are horizontally scaled externally to the sequencer through a proxy, confiugration must be readable by these hosts & the design space for this rate limit should be explored -- i.e globally synchronized rate limit vs per-host limits.
+        Rate limit how many conditional txs can be sent over a configured time period. Like with the runtime shutoff, if the validation proxy is horizontally scaled, the design space for this rate limit should be explored -- i.e globally synchronized rate limit vs per-host limits.
+
+        When this rate limit is reached, the proxy should set the response HTTP status code to 429 (Too Many Requests), as an indicator for the caller.
+
+
+With the validation proxy described in (2) and the `sendRawTransactionConditional` op-geth endpoint described in (1), the cloud architecture for how this setup works should look like the following diagram. An operator that wishes to expose the endpoint publicly without the validating proxy in front can simply route the method directly to the sequencer from the ingress router.
+
+![Cloud Architecture](cloud-architecture.png)
+
+With this initial set of validation rules, we should be in a good position to safely launch this endpoint permissionlessly. Instrumentation is critical to surface further action -- key metrics to track:
+
+* **conditional requests / s (by caller)**: understanding of overall usage and alert setup with the configured rate limits. The breakdown of requests by caller will help identify single-use authentication keys and the general activity of different bundlers
```

Comment:
Just need to be careful about this because the caller is untrusted input and a naive implementation of metrics could allow for somebody to trigger an oom on the service

Reply:
yea good callout